### ISSUE TITLE
Rule Change: prevent underscore dangles in object properties

### ISSUE THREAD

**Issue by thibaudszy** [eslint/eslint#17453]
### What rule do you want to change?

no-underscore-dangle

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
const a = { _b: "b" }; // no error

console.log(a)
```

### What does the rule currently do for this code?

Nothing

### What will the rule do after it's changed?

Flag this code as a `no-underscore-dangle` violation

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

I already implemented this as a separate rule in a private plugin but I believe it could benefit all users of ESLint. As it is implemented, the rule allows to ban all dangling underscores for object properties, with some exceptions.

**Comment by aladdin-add** [comment]
Thanks for the input. As [stated in the README](https://github.com/eslint/eslint#stylistic-rule-updates), we actually aren't making any further updates to rules that enforce stylistic preferences.
